src/transactions/test/SetTrustLineFlagsTests.cpp

```diff
@@ -287,3 +288,767 @@ TEST_CASE("set trustline flags", "[tx][settrustlineflags]")
+                        changeTrustAssetToTrustLineAsset(shareNative1)));
+
+                    checkLiquidityPool(*app, poolNative1, 50, 200, 100, 1);
+
+                    auto cur1Tl = acc1.loadTrustLine(cur1);
+                    REQUIRE(getTrustLineEntryExtensionV2(cur1Tl)
+                                .liquidityPoolUseCount == 0);
+
+                    REQUIRE(acc1.getTrustlineBalance(cur1) == 0);
+                    acc1.claimClaimableBalance(cur1BalanceID);
+                    REQUIRE(acc1.getTrustlineBalance(cur1) == 200);
+
+                    int64_t balance = acc1.getBalance();
+                    acc1.claimClaimableBalance(nativeBalanceID);
+                    REQUIRE(acc1.getBalance() == balance - 100 + 50);
+                }
+            }
+
+            SECTION("validate op num and tx account is used in hash")
+            {
+                acc1.changeTrust(cur1, 200);
+                root.pay(acc1, cur1, 200);
+
+                acc1.changeTrust(shareNative1, 100);
+
+                acc1.liquidityPoolDeposit(poolNative1, 50, 200, Price{1, 4},
+                                          Price{1, 4});
+
+                checkLiquidityPool(*app, poolNative1, 50, 200, 100, 1);
+
+                // Allow acc1 to submit an op from root
+                auto sk1 = makeSigner(acc1, 100);
+                root.setOptions(setSigner(sk1));
+
+                auto revokeOp =
+                    flagOp == TrustFlagOp::ALLOW_TRUST
+                        ? root.op(allowTrust(acc1, cur1, 0))
+                        : root.op(setTrustLineFlags(
+                              acc1, cur1,
+                              clearTrustLineFlags(AUTHORIZED_FLAG)));
+                applyCheck(acc1.tx({root.op(payment(acc1, 1)), revokeOp}),
+                           *app);
+
+                // acc1 is the txSource account, which is what should be used
+                // for the balanceID
+                auto rootNativeBalanceID =
+                    getRevokeBalanceID(root, native, poolNative1, 1);
+                auto acc1NativeWrongIndexBalanceID =
+                    getRevokeBalanceID(acc1, native, poolNative1, 0);
+                auto acc1Cur2BalanceID =
+                    getRevokeBalanceID(acc1, cur2, poolNative1, 1);
+                auto acc1NativeBalanceID =
+                    getRevokeBalanceID(acc1, native, poolNative1, 1);
+
+                REQUIRE_THROWS_AS(
+                    acc1.claimClaimableBalance(rootNativeBalanceID),
+                    ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+                REQUIRE_THROWS_AS(
+                    acc1.claimClaimableBalance(acc1NativeWrongIndexBalanceID),
+                    ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+                REQUIRE_THROWS_AS(acc1.claimClaimableBalance(acc1Cur2BalanceID),
+                                  ex_CLAIM_CLAIMABLE_BALANCE_DOES_NOT_EXIST);
+
+                acc1.claimClaimableBalance(acc1NativeBalanceID);
```

Comment:
What about similar checks for the non-native asset?

Reply:
Fixed